crates/store/src/db/mod.rs

```diff
@@ -406,7 +407,10 @@ impl Db {
             .get()
             .await
             .map_err(DatabaseError::MissingDbConnection)?
-            .interact(move |conn| sql::get_note_sync(conn, block_num, &note_tags))
+            .interact(move |conn| {
+                let transaction = conn.transaction().map_err(DatabaseError::SqliteError)?;
```

Comment:
This `.map_err` breaks a bit the pattern used in the previous methods (`conn.transaction()?`). Is there a reason why this methods returns an explicit error?